File: project_euler/Problem031/Problem031.go
package project_euler

import "fmt"

func Problem031() {
    target := 200
    coins := []int{1, 2, 5, 10, 20, 50, 100, 200}
    ways := make([]int, target + 1)
    for i := 1; i <= target; i++ {
        ways[i] = 0
    }
    ways[0] = 1
    for _, coin := range coins {
        for i := coin; i <= target; i++ {
            ways[i] += ways[i - coin]
        }
    }
    fmt.Println(ways[target])
}

File: project_euler/Problem031/Problem031.py
class Problem031:
    def solution(self) -> None:
        target, coins = 200, [1, 2, 5, 10, 20, 50, 100, 200]
        ways = [0 for _ in range(target + 1)]
        ways[0] = 1
        for coin in coins:
            for i in range(coin, target + 1):
                ways[i] += ways[i - coin]
        print(ways[target])
